tools/acrn-crashlog/usercrash/README.rst

```diff
@@ -53,19 +53,11 @@ The work flow diagram:
 Usage
 *****
 
-- The server is launched automatically at boot, and the client is configured in
-  ``core_pattern`` or ``coredump-wrapper``. In ``prepare.service``, it will
-  check the content of ``/proc/sys/kernel/core_pattern``. If there is
-  ``coredump-wrapper``, which means that ``core_pattern`` has been set in
-  ``systemd``, no need to do it again. Otherwise, the content should be
-  changed by:
-
-.. code-block:: none
-
-   $ echo "|/usr/bin/usercrash_c %p %e %s" > /proc/sys/kernel/core_pattern
-
-That means client will be triggered once userspace crash occurs. Then the
-event will be sent to server from client.
+- The server is launched automatically at boot after this tool is enabled, and
+  the client is configured in ``usercrash-wrapper``, which is set as the app
```

Comment:
Can you explain a little more what you mean by `the client is configured in usercrash-wrapper`, is this a tool to be run or a config file to be edited?

Reply:
There is a config file 80-coredump.conf, the content likes the following:
#  This file is part of acrn-hypervisor.

# See sysctl.d(5) for the description of the files in this directory,
# and systemd-coredump(8) and core(5) for the explanation of the
# setting below.

kernel.core_pattern=|/usr/bin/usercrash-wrapper %E %P %u %g %s %t %c %h %e %p %i %I %d

It's used to replace the default core_pattern. 

In the end of usercrash-wrapper, it will invoke the usercrash_c(client) and the default coredump application separately once a crash occurs in user space:
tee >(/usr/bin/usercrash_c ${usercrash_var}) | ${default_app} ${default_var}